Client/Coordinators/SettingsCoordinator.swift

```diff
@@ -97,6 +99,7 @@ class SettingsCoordinator: BaseCoordinator,
                 referringPage: .settings,
                 profile: profile
             )
+            (viewController as? FirefoxAccountSignInViewController)?.qrCodeNavigationHandler = self
```

Comment:
Can we try to make the `FirefoxAccountSignInViewController.getSignInOrFxASettingsVC` static function returns the `FirefoxAccountSignInViewController` type directly instead of `UIViewController`? Would be nicer to set the `qrCodeNavigationHandler` that way

Reply:
yes indeed, but the function can return either a `FirefoxAccountSignInViewController` or `SyncContentSettingsViewController` depending on some state

Reply:
Ah I see. Well this is annoying. Let's leave it be for now then